### PR TITLE
FIX-#4185: Fix rolling across column partitions.

### LINKED ISSUE #4185: DataFrame.rolling() gives incorrect result when windows span two column partitions and axis == 1
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS Monterey 12.2
- **Modin version** (`modin.__version__`): 0.13.0+19.g62d73b78
- **Python version**: 3.9.9
- **Code we can use to reproduce**: 

```python
import modin.pandas as pd
df = pd.DataFrame([range(33)])
# result is nan
print(df.rolling(window=2, axis=1).min().iloc[0, -1])
import pandas
pdf = pandas.DataFrame([range(33)])
# result is 33
print(pdf.rolling(window=2, axis=1).min().iloc[0, -1])
```

### Describe the problem

When `axis=1`, we "fold" (really a better term is "map full axis partitions") along axis=0 instead of axis=1. The result is that each column partition gets results independently of other column partitions. We need to pass `axis` to the fold calls as [here](https://github.com/modin-project/modin/blob/62d73b7882fc58c4ef88f05bf596ece9f3cfe4fe/modin/core/storage_formats/pandas/query_compiler.py#L1076).

### PR DESCRIPTION
**Pull request by mvashishtha**
## What do these changes do?

Fix rolling across column partitions by mapping axis partitions that span the correct dimension.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4185
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/core/storage_formats/base/doc_utils.py [hunk 1]
```diff
@@ -567,10 +567,11 @@ def doc_window_method(
     callable
     """
     template = """
-        Create {win_type} and {action} for each window.
+        Create {win_type} and {action} for each window over the given axis.
 
         Parameters
         ----------
+        fold_axis : {{0, 1}}
```
Comment (naren-ponder) [inline modin/core/storage_formats/base/doc_utils.py:574]: Perhaps I'm missing something, but don't see `fold_axis` as a parameter for `doc_window_method`.

Comment (mvashishtha) [reply modin/core/storage_formats/base/doc_utils.py:574]: It's not a parameter to `doc_window_method`, but it's a parameter to each of the window methods in the query compiler :) `doc_window_method` generates the docstrings for those methods. The `fold_axis` line here is in a docstring template and will go into the docstring for each method.

File: modin/core/storage_formats/base/doc_utils.py [hunk 2]
```diff
@@ -286,7 +286,7 @@ def doc_reduce_agg(method, refer_to, params=None, extra_params=None):
 
     Parameters
     ----------
-    axis : {{0, 1}}
+    fold_axis : {{0, 1}}
```
[the remark this answers is not in the document]
Comment (mvashishtha) [reply modin/core/storage_formats/base/doc_utils.py:289]: The fold is basically a `map_axis_partitions`, which maps an operations across either entire rows or entire columns. Mapping across both at the same time would be equivalent to mapping across the entire dataframe, which the algebra has no support for.

File: modin/core/dataframe/algebra/fold.py [hunk 3]
```diff
@@ -20,29 +20,26 @@ class Fold(Operator):
...
-            Specifies axis to apply function along.
 
         Returns
         -------
         callable
             Function that takes query compiler and executes Fold function.
         """
 
-        def caller(query_compiler, *args, **kwargs):
+        def caller(query_compiler, fold_axis=None, *args, **kwargs):
```
Comment (devin-petersohn) [inline modin/core/dataframe/algebra/fold.py:38]: Let's add a doc for the new parameter.

**Review by devin-petersohn — COMMENTED** [review]
Looks pretty good, left a comment.

File: modin/core/dataframe/algebra/fold.py (hunk 3 above)
Comment (mvashishtha) [reply modin/core/dataframe/algebra/fold.py:38]: This inner function didn't have a docstring at all, but I added it, and documented the new parameter.

Comment (mvashishtha) [reply modin/core/dataframe/algebra/fold.py:38]: Done.

File: modin/core/dataframe/algebra/fold.py [hunk 4]
```diff
@@ -20,29 +20,46 @@ class Fold(Operator):
...
-            _axis = kwargs.get("axis") if axis is None else axis
+        def caller(query_compiler, fold_axis=None, *args, **kwargs):
+            """
+            Execute Fold function against passed query compiler.
+
+            Parameters
+            ----------
+            query_compiler : BaseQueryCompiler
+                The query compiler to execute the function on.
+            axis : int, optional
```
Comment (naren-ponder) [inline modin/core/dataframe/algebra/fold.py:46]: This should be `fold_axis` I think.

**Review by naren-ponder — DISMISSED** [review]
Left one minor comment, but LGTM to me otherwise!

File: modin/core/dataframe/algebra/fold.py (hunk 4 above)
Comment (mvashishtha) [reply modin/core/dataframe/algebra/fold.py]: Thanks. Fixed.

**Review by naren-ponder — APPROVED** [review]
🚢

**Review by devin-petersohn — APPROVED** [review]
LGTM, thanks @mvashishtha !

**Pull request merged**